Add missing features to the 'document' page. (#20305)
mdn/content: files/en-us/web/api/document/index.md

  - : Gets/sets directionality (rtl/ltr) of the document.
- {{DOMxRef("Document.domain")}} {{Deprecated_Inline}}
  - : Gets/sets the domain of the current document.

> - {{DOMxRef("Document.fragmentDirective")}} {{ReadOnlyInline}} {{Experimental_Inline}}
>   - : This object may be used to expose additional information about the text fragment or other fragment directives in the future.

Reviewer: I don't think we should add non-documented features to this list. Adding a lot of red links
 doesn't make the docs any better.
 
Forgotten documented features are ok.

```suggestion
```

  - : Sets or gets the title of the current document.
- {{DOMxRef("Document.URL")}} {{ReadOnlyInline}}
  - : Returns the document location as a string.

> - {{DOMxRef("Document.wasDiscarded")}} {{ReadOnlyInline}} {{Experimental_Inline}}
>   - : Indicates whether the page was discarded while in a hidden tab.

Reviewer: ```suggestion
```

  - : Returns an array of all elements at the specified coordinates.
- {{DOMxRef("Document.enableStyleSheetsForSet()")}} {{Deprecated_Inline}} {{Non-standard_Inline}}
  - : Enables the style sheets for the specified style sheet set.
- {{DOMxRef("Document.exitFullscreen()")}}

>   - : Stops document’s fullscreen element from being displayed fullscreen.

Author: ```suggestion
  - : Stops document's fullscreen element from being displayed fullscreen.
```

  - : Fired when a static {{HTMLElement("script")}} element finishes executing its script
- {{DOMxRef("Document.beforescriptexecute_event", "beforescriptexecute")}} {{Non-standard_Inline}}
  - : Fired when a static {{HTMLElement("script")}} is about to start executing.

> - {{DOMxRef("SecurityPolicyViolationEvent", "securitypolicyviolation")}}
>   - : Fired when content security policy is violated.

Reviewer: This is not the correct link:
```suggestion
- {{DOMxRef("Document.securitypolicyviolation_event", "securitypolicyviolation")}}
  - : Fired when a content security policy is violated.
```

- {{DOMxRef("Document/keyup_event", "keyup")}}
  - : Fired when a key is released.

### Lifecycle events

- {{DOMxRef("Document/freeze_event", "freeze")}} {{Experimental_Inline}}
  - : Fired when the document transitions to `FROZEN` state.
- {{DOMxRef("Document/resume_event", "resume")}} {{Experimental_Inline}}
  - : Fired when the document transitions from `FROZEN` to `ACTIVE` state.

> 

Reviewer: Red links. (And the text look weird, I've never seen capitalized keyword like this).
```suggestion
```

Author: > I've never seen capitalized keyword like this

These are the actual state values in the specification. So I thought mentioning them as it is will be better.

> ### Lifecycle events
> 
> - {{DOMxRef("Document/freeze_event", "freeze")}} {{Experimental_Inline}}
>   - : Fired when the document transitions to `FROZEN` state.
> - {{DOMxRef("Document/resume_event", "resume")}} {{Experimental_Inline}}  
>   - : Fired when the document transitions from `FROZEN` to `ACTIVE` state.
> 

Reviewer: ```suggestion
```

Red links…
